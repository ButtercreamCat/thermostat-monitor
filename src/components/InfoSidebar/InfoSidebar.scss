@import "../../sass/abstracts/variables";

.unitInfo_sidebar {
  width: 20%;
  height: 100%;
  position: relative;
  border-right: 1px solid $grey;

  &_header {
    position: relative;
    margin-bottom: 40px;
    width: 100%;
    // border-radius: 8px;
    // box-shadow: $box-shadow;
  }

  &_header_pic {
    width: 100%;
    display: block;
    top: 0;
    left: 0;
    // border-radius: 8px;
  }

  &_header_overlay {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.4);
    // border-radius: 8px;
  }

  &_header_title {
    width: 100%;
    text-align: center;
    position: absolute;
    top: 50%;
    left: 0%;
    transform: translate(0%, -100%);
    color: $white;
    font-size: 3rem;
  }

  &_btns {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 100px;
    a {
      margin: 20px 0;
    }
  }

  &_logo {
    position: absolute;
    bottom: 30px;
    left: 50%;
    transform: translate(-55%, 0);
    max-width: 100px;
  }
}

@media only screen and (max-width: 768px) {
  .unitInfo_sidebar {
    width: 100%;
    height: auto;
    position: absolute;
    top: 0;
    left: 0;

    .button-withoutBorder-withIcon {
      margin: 20px 0 10px 16px;
    }

    &_btns {
      flex-direction: row;
      align-items: center;
      justify-content: center;
      margin-top: 0px;
      a {
        margin: 20px;
      }
    }

    &_header,
    &_logo {
      display: none;
    }
  }
}
