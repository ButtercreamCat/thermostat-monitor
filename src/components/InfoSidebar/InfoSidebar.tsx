import React, { Component } from "react";
import "./InfoSidebar.scss";
import { Link } from "react-router-dom";

interface InfoSidebarProps {
  id: string;
  url: string;
}

class InfoSidebar extends Component<InfoSidebarProps> {
  state = {
    section: "Thermostat"
  };

  render() {
    const { id, url } = this.props;
    const { section } = this.state;
    return (
      <section className="unitInfo_sidebar">
        <Link to="/" className="button button-withoutBorder-withIcon">
          <span className="icon-back"></span>
          Back
        </Link>
        <div className="unitInfo_sidebar_header">
          <img
            src="/images/room.jpg"
            alt="header"
            className="unitInfo_sidebar_header_pic"
          ></img>
          <div className="unitInfo_sidebar_header_overlay"></div>
          <h1 className="unitInfo_sidebar_header_title">Unit {id}</h1>
        </div>
        <div className="unitInfo_sidebar_btns">
          <Link
            to={`${url}/thermostat`}
            className={`button button-rect-withIcon ${
              section === "Thermostat" ? "button-active" : ""
            }`}
            onClick={() => {
              this.setState({ section: "Thermostat" });
            }}
          >
            <span className="icon-thermostat"></span>
            <div>Thermostat</div>
          </Link>
          <Link
            to={`${url}/chart`}
            className={`button button-rect-withIcon ${
              section === "Chart" ? "button-active" : ""
            }`}
            onClick={() => {
              this.setState({ section: "Chart" });
            }}
          >
            <span className="icon-chart"></span>
            <div>Chart</div>
          </Link>
        </div>
        <img
          className="unitInfo_sidebar_logo"
          src="/images/logo.svg"
          alt="logo"
        ></img>
      </section>
    );
  }
}

export default InfoSidebar;
