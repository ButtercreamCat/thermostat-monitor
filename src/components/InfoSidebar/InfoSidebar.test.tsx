import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import InfoSidebar from "./InfoSidebar";

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

beforeEach(() => {
  wrapper = shallow(<InfoSidebar id="101" url="/unit/101"></InfoSidebar>);
});

test("should render InfoSidebar correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should display correct unit number", () => {
  expect(wrapper.find(".unitInfo_sidebar_header_title").text()).toBe(
    "Unit 101"
  );
});

test("should highlight button on click", () => {
  wrapper
    .find(".button-rect-withIcon")
    .at(0)
    .simulate("click");
  expect(
    wrapper
      .find(".button-rect-withIcon")
      .at(0)
      .hasClass("button button-rect-withIcon button-active")
  ).toBe(true);
});

test("should change state on click", () => {
  wrapper
    .find(".button-rect-withIcon")
    .at(0)
    .simulate("click");
  expect(wrapper.state("section")).toBe("Thermostat");
  wrapper
    .find(".button-rect-withIcon")
    .at(1)
    .simulate("click");
  expect(wrapper.state("section")).toBe("Chart");
});
