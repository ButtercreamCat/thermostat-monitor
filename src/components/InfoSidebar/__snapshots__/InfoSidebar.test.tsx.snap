// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InfoSidebar correctly 1`] = `
<section
  className="unitInfo_sidebar"
>
  <Link
    className="button button-withoutBorder-withIcon"
    to="/"
  >
    <span
      className="icon-back"
    />
    Back
  </Link>
  <div
    className="unitInfo_sidebar_header"
  >
    <img
      alt="header"
      className="unitInfo_sidebar_header_pic"
      src="/images/room.jpg"
    />
    <div
      className="unitInfo_sidebar_header_overlay"
    />
    <h1
      className="unitInfo_sidebar_header_title"
    >
      Unit 
      101
    </h1>
  </div>
  <div
    className="unitInfo_sidebar_btns"
  >
    <Link
      className="button button-rect-withIcon button-active"
      onClick={[Function]}
      to="/unit/101/thermostat"
    >
      <span
        className="icon-thermostat"
      />
      <div>
        Thermostat
      </div>
    </Link>
    <Link
      className="button button-rect-withIcon "
      onClick={[Function]}
      to="/unit/101/chart"
    >
      <span
        className="icon-chart"
      />
      <div>
        Chart
      </div>
    </Link>
  </div>
  <img
    alt="logo"
    className="unitInfo_sidebar_logo"
    src="/images/logo.svg"
  />
</section>
`;
